core/node/rpc/metrics_interceptor.go

```diff
@@ -74,11 +92,37 @@ func (i *metricsInterceptor) WrapStreamingHandler(next connect.StreamingHandlerF
 		ctx context.Context,
 		conn connect.StreamingHandlerConn,
 	) error {
-		m := i.openClientStreams.With(prometheus.Labels{"proc": conn.Spec().Procedure})
+		var (
+			proc = conn.Spec().Procedure
+			m    = i.openClientStreams.With(prometheus.Labels{"method": proc})
+			s, _ = i.serverStreamsStatusCode.CurryWith(prometheus.Labels{"method": proc})
+		)
 
 		m.Inc()
 		defer m.Dec()
 
-		return next(ctx, conn)
+		err := next(ctx, conn)
+
+		s.With(prometheus.Labels{"status": errorToStatus(err)}).Inc()
+
+		return err
 	}
 }
+
+func errorToStatus(err error) string {
+	if err == nil {
+		return "success"
+	}
+
+	var connectErr *connect.Error
+	if ok := errors.As(err, &connectErr); ok {
+		return connectErr.Code().String()
+	}
+
+	var riverErr *base.RiverErrorImpl
```

Comment:
river error needs to come first: at this stage usually it's connect error which wraps river error, so first if will always succeed (river error code is superset of grpc code and is more detailed).